### PR TITLE
Remove connections for a node that is being deleted.

### LINKED ISSUE #7135: Removing a node does not remove the reference to it from connected nodes
Like shown on the video:

https://github.com/enso-org/enso/assets/1436948/dbaa3f52-21b9-4ee1-93f1-341d22d57c2d

This seems like a regression - IIRC in the past when I removed a node, the connections are disconnected and the reference in other nodes would be removed or replaced with a `_` placeholder.

This is especially problematic (as in the video) when the node was connected to an optional argument. If the argument disappeared (as I recall used to be the case in the past) - the node would come back to the previous setting. Now, instead, the node becomes erroneous because it references a name that no longer exists.

### PR DESCRIPTION
**Pull request by MichaelMauderer**
### Pull Request Description

Fixes https://github.com/enso-org/enso/issues/7135

https://github.com/enso-org/enso/assets/1428930/85433a69-03dc-4ad9-9563-717a78ff6ff8

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: app/gui/src/controller/graph/executed.rs [hunk 1]
```diff
@@ -451,6 +451,25 @@ impl Handle {
...
+        } else {
+            // Failure has no good way to propagate multiple errors with `Failure`. So we propagate
+            // only the first one.
+            let first_error = errors
+                .into_iter()
+                .next()
+                .expect("Errors are not empty, so there must be a first element.");
+            Err(first_error)
+        }
+    }
```
Comment (kazcw) [inline app/gui/src/controller/graph/executed.rs]: why not `if let Some(error) = errors.into_iter().first()`?

**Review by kazcw — APPROVED** [review]

**Review by farmaazon — APPROVED** [review]

**Review by wdanilo — APPROVED** [review]

File: app/gui/src/controller/graph/executed.rs (hunk 1 above)
Comment (MichaelMauderer) [reply app/gui/src/controller/graph/executed.rs]: Much nicer! Unfortunately, `first` only exists on `Vec` not on `IntoIter` so I'll have to keep that. Or do a `take`, which will mutate the `Vec` and thus is not really better.

Comment (kazcw) [reply app/gui/src/controller/graph/executed.rs]: Oh of course, `next`

**Comment by kazcw** [comment]
# QA: :green_circle:

**Pull request merged**